metrics/transport.go

```diff
@@ -119,7 +119,7 @@ func PullMetrics(ctx context.Context, metricsPort int, actions []string) error {
 	go func() {
 		<-ctx.Done()
 		//nolint:staticcheck
-		if err := srv.Shutdown(nil); err != nil {
+		if err := srv.Shutdown(context.Background()); err != nil {
```

Comment:
According to docs the Shutdown can potentially hang indefinitely. We should probably have something like:
context.WithTimeout(context.Background(), 5*time.Second)